multisense_ros/include/multisense_ros/camera.h

```diff
@@ -333,17 +333,20 @@ class Camera {
     //
     // Diagnostics
     diagnostic_updater::Updater diagnostic_updater_;
-    void deviceInfoDiagnostic(diagnostic_updater::DiagnosticStatusWrapper &stat);
-    void deviceStatusDiagnostic(diagnostic_updater::DiagnosticStatusWrapper &stat);
+    void deviceInfoDiagnostic(diagnostic_updater::DiagnosticStatusWrapper& stat);
+    void deviceStatusDiagnostic(diagnostic_updater::DiagnosticStatusWrapper& stat);
+    void ptpStatusDiagnostic(diagnostic_updater::DiagnosticStatusWrapper& stat);
 
-    void diagnosticTimerCallback(const ros::TimerEvent &);
+    void diagnosticTimerCallback(const ros::TimerEvent&);
     ros::Timer diagnostic_trigger_;
 
     //
-    // Timesync settings
+    // Timestamping and timesync settings
+    ros::Time convertImageTimestamp(uint32_t time_secs, uint32_t time_microsecs);
+
     bool ptp_time_sync_ = false;
     bool network_time_sync_ = false;
-
+    std::atomic_bool ptp_status_;
```

Comment:
at first i thought this mean ptp_initialized_, but it looks like it is ptp_value_updated_ or something?
think it should be renamed to clarify intended use